decision_service/dsclientlib/ds_config_collection.cc

```diff
@@ -0,0 +1,22 @@
+#include "ds_config_collection.h"
+
+namespace decision_service { namespace utility {
+
+  void config_collection::set(const char* name, const char* value) {
+    _map[name] = value;
+  }
+
+  const char* config_collection::get(const char* name, const char* defval) const {
+    const auto it = _map.find(name);
+    if (it != _map.end()) 
+      return it->second.c_str();
+    return defval;
+  }
+
+  int config_collection::get_int(const char* name, int defval) const {
+    const auto it = _map.find(name);
+    if (it != _map.end())
+      return std::stoi(it->second);
```

Comment:
>std::stoi [](start = 13, length = 9)

I had some issues that this doesn't work on all compilers

Reply:
switched to atoi

---
In reply to: [186734171](https://github.com/JohnLangford/vowpal_wabbit/pull/1479#discussion_r186734171) [](ancestors = 186734171)